optimade/models/jsonapi.py

```diff
@@ -180,6 +180,13 @@ class Relationships(BaseModel):
         id
     """
 
+    id: Optional[Any] = Schema(..., description="Not allowed key")
+    type: Optional[Any] = Schema(..., description="Not allowed key")
+
+    @validator("id", "type")
+    def check_illegal_relationships_fields(cls, v):
+        raise AssertionError('"id", "type" MUST NOT be fields under relationships')
```

Comment:
Should we also change all these other validators to `ValueError`?

Reply:
To speed things up, I'll accept as is and you can merge if you want. If you change it just re-request my review and I'll accept.

Reply:
I guess we should? Since we didn't encounter this before, it must mean that we are not actually testing this validator?